src/field/field3d.cxx

```diff
@@ -1065,8 +1128,18 @@ void shiftZ(Field3D &var, int jx, int jy, double zangle) {
 }
 
 void shiftZ(Field3D &var, double zangle, REGION rgn) {
-  for (const auto &i : var.region2D(rgn))
-    shiftZ(var, i.x, i.y, zangle);
+  const auto region_str = REGION_STRING(rgn);
+
+  // Only allow a whitelist of regions for now
+  ASSERT2(region_str == "RGN_ALL" || region_str == "RGN_NOBNDRY" ||
+          region_str == "RGN_NOX" || region_str == "RGN_NOY");
+
+  const Region<Ind2D> &region = var.getMesh()->getRegion2D(region_str);
+
+  // Could be OpenMP if shiftZ(Field3D, int, int, double) didn't throw
```

Comment:
Since an exception thrown would most likely just terminate the program, can we just allow it to happen. It makes debugging more difficult, but could be done by turning off OpenMP.